x-pack/filebeat/input/entityanalytics/provider/azuread/azure_test.go

```diff
@@ -36,13 +36,18 @@ func TestAzure_DoFetch(t *testing.T) {
 
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
-	got, err := a.doFetch(ctx, ss, false)
+	gotUsers, gotDevices, err := a.doFetch(ctx, ss, false)
 	require.NoError(t, err)
 
 	var wantModifiedUsers collections.UUIDSet
 	for _, v := range mockfetcher.UserResponse {
 		wantModifiedUsers.Add(v.ID)
 	}
+	var wantModifiedDevices collections.UUIDSet
+	for _, v := range mockfetcher.DeviceResponse {
+		wantModifiedDevices.Add(v.ID)
+	}
 
-	require.Equal(t, wantModifiedUsers.Values(), got.Values())
+	require.Equal(t, wantModifiedUsers.Values(), gotUsers.Values())
+	require.Equal(t, wantModifiedDevices.Values(), gotDevices.Values())
 }
```

Comment:
A lot of conditional blocks are introduced in `x-pack/filebeat/input/entityanalytics/provider/azuread/azure.go` but unit testing has been very limited. Can we add more unit tests here

Reply:
AFAICS the proportion of testing has remained essentially unchanged with this addition.

Reply:
Found an area that wanted tests added, added input state to make the current tests cover it. The remainder should be covered by integration tests when the integration is a implemented.